### PR TITLE
Add StackReferenceArgs.Builder

### LINKED ISSUE #534: StackReferenceArgs missing `T` overload
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
Missed adding an overload to StackReferenceArgs.

StackReference can accept a plain string if the reference is passed as the first argument, but if the Args builder is used it only takes an output. Should be an easy fix.

https://github.com/pulumi/pulumi-java/blob/main/sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReferenceArgs.java

### PR DESCRIPTION
**Pull request by pawelprazak**
# Description

Fixes #534

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReferenceArgs.java [hunk 1]
```diff
@@ -7,26 +7,79 @@
...
+     */
+    public static final class Builder {
+        @Nullable
+        private Output<String> name;
+
+        /**
+         * @param name the name of the stack to reference.
+         * @return the {@link Builder}
+         * @see StackReferenceArgs#name()
+         */
```
Comment (dixler) [inline sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReferenceArgs.java:62]: Nit: would like something describing the structure of the string `<organization>/<project>/<stack>`

**Review by dixler — APPROVED** [review]
Aside from the nitpick, it works on my machine.

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
 ### Improvements
 
 ### Bug Fixes
+
+- Fix #534: add StackReferenceArgs.builder() and StackReferenceArgs.Builder, deprecate StackReferenceArgs.getName()
```
Comment (t0yv0) [inline CHANGELOG_PENDING.md:5]: This sounds like an Improvement not a Bug Fix.

**Review by t0yv0 — APPROVED** [review]

File: sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReferenceArgs.java (hunk 1 above)
Comment (Frassle) [reply sdk/java/pulumi/src/main/java/com/pulumi/resources/StackReferenceArgs.java:62]: It's actually <stack> or <org>/<stack> or <org>/<proj>/<stack> just to complicate things :)

File: CHANGELOG_PENDING.md (hunk 2 above)
Comment (pawelprazak) [reply CHANGELOG_PENDING.md]: thank you, moved

**Pull request merged**